container-images/cuda/Containerfile

```diff
@@ -46,16 +45,14 @@ ARG HUGGINGFACE_HUB_VERSION=0.26.2
 ARG OMLMD_VERSION=0.1.6
 
 # Install minimal runtime dependencies
-RUN dnf install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-9.noarch.rpm && \
-    dnf install -y python3 python3-pip && dnf clean all && rm -rf /var/cache/*dnf*
+RUN dnf install -y python3 python3-pip nvidia-driver-cuda-libs && \
```

Comment:
I had to add this for this lib:

/lib64/libcuda.so.1

```
$ ldd /usr/bin/llama-cli
	linux-vdso.so.1 (0x00007f219dc64000)
	libllama.so => /lib64/libllama.so (0x00007f219dae4000)
	libggml.so => /lib64/libggml.so (0x00007f2188000000)
	libstdc++.so.6 => /lib64/libstdc++.so.6 (0x00007f2187c00000)
	libm.so.6 => /lib64/libm.so.6 (0x00007f219da09000)
	libgcc_s.so.1 => /lib64/libgcc_s.so.1 (0x00007f219d9ee000)
	libc.so.6 => /lib64/libc.so.6 (0x00007f2187800000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f219dc66000)
	libcudart.so.12 => /usr/local/cuda/targets/x86_64-linux/lib/libcudart.so.12 (0x00007f2187400000)
	libcublas.so.12 => /usr/local/cuda/targets/x86_64-linux/lib/libcublas.so.12 (0x00007f2180a00000)
	libcublasLt.so.12 => /usr/local/cuda/targets/x86_64-linux/lib/libcublasLt.so.12 (0x00007f215f000000)
	libcuda.so.1 => /lib64/libcuda.so.1 (0x00007f215c000000)
	libgomp.so.1 => /lib64/libgomp.so.1 (0x00007f219d9a5000)
	libdl.so.2 => /lib64/libdl.so.2 (0x00007f219d99e000)
	libpthread.so.0 => /lib64/libpthread.so.0 (0x00007f219d999000)
	librt.so.1 => /lib64/librt.so.1 (0x00007f219d994000)
```

which makes me wonder, how was this working before without explicitly installing that.

Reply:
So originally the container would use libllama.so and libggml.so from the build of llama.cpp with cuda. I added those files to /usr/lib and updated the linker to point to those instead (following the example from the official dockerfile). 

Even before that I already had those packages installed in the container. Its most likely due to cuda container toolkit passing the library from the nvidia driver on the host to the container.

Reply:
Should we remove nvidia-driver-cuda-libs? Is it breaking nvidia-container-toolkit?

Reply:
I wonder can we CI some of this stuff even without Nvidia hardware in the CI runners